wdl/BatchEvidenceMerging.wdl

```diff
@@ -110,6 +127,34 @@ task MergeEvidence {
     mv ~{write_lines(files)} evidence.list
     mv ~{write_lines(samples)} samples.list
 
+    # For legacy evidence files that were not dictionary sorted, removing non-primary contigs fixes the GATK error
+    if ~{subset_primary_contigs} || ~{rename_samples}; then
+      mkdir evidence
+      touch evidence.tmp
+      cut -f1 ~{primary_contigs_fai} > contigs.list
+      while read fil sample; do
+        FILENAME=$(basename $fil)
+        OUT="evidence/$FILENAME"
+        if [[ "~{evidence}" == "pe" ]]; then
+          zcat $fil \
+            | awk -F'\t' -v OFS='\t' -v SAMPLE="$sample" '!second_file{chroms[$1]; next} {~{if subset_primary_contigs then "if ($1 in chroms && $4 in chroms)" else ""} print ~{if rename_samples then "$1,$2,$3,$4,$5,$6,SAMPLE" else "$0"} }' contigs.list second_file=1 - \
+            | bgzip > $OUT
+        elif [[ "~{evidence}" == "sr" ]]; then
+          zcat $fil \
+            | awk -F'\t' -v OFS='\t' -v SAMPLE="$sample" '!second_file{chroms[$1]; next} {~{if subset_primary_contigs then "if ($1 in chroms)" else ""} print ~{if rename_samples then "$1,$2,$3,$4,SAMPLE" else "$0"} }' contigs.list second_file=1 - \
+            | bgzip > $OUT
+        else
+          # baf - also uniquify records from old files
+          zcat $fil \
+            | awk -F'\t' -v OFS='\t' -v SAMPLE="$sample" '!second_file{chroms[$1]; next} {~{if subset_primary_contigs then "if ($1 in chroms)" else ""} print ~{if rename_samples then "$1,$2,$3,SAMPLE" else "$0"} }' contigs.list second_file=1 - \
+            | awk -F'\t' -v OFS='\t' '!_[$1"_"$2]++' - \
```

Comment:
I don't know the background on this but this seems like a bit of a hidden thing to have happen if one of these two options is specified. I could imagine someone getting pretty confused if they ever had to debug an issue that revolved around duplicate records. Could we always do this if BAF is specified? Or make an additional option to do it?

Reply:
Or at least put in an explicit comment / warning somewhere, if we don't want to go back and re-test?

Reply:
Added another comment at the top about this